File: 3R/10405_JollyJumpers.cpp
#include <iostream>
#include <set>
#include <cstdlib>

using namespace std;

int main(){
    int n;
    while(cin >> n){
        int a; cin >> a;
        set<int> jset;
        for(int i=1; i<n; i++){
            int b; cin >> b;
            int d = abs(a-b);
            if(d && d<n) jset.insert(d);
            a = b;
        }
        if(jset.size() == n-1) cout << "Jolly";
        else cout << "Not jolly";
        cout << endl;
    }
    return 0;
}

File: 3R/10405_JollyJumpers.java
import java.util.*;

class Main{
	public static void main(String[] args){
		Scanner sc = new Scanner(System.in);
        while(sc.hasNext()){
            int n = sc.nextInt();
            int a = sc.nextInt();
            
            Set<Integer> jset = new HashSet<Integer>();
            
            for(int i=1; i<n; i++){
                int b = sc.nextInt();
                int d = Math.abs(a-b);

                if(d>0 && d<n) jset.add(d);
                a = b;
            }

            System.out.println(jset.size()==n-1? "Jolly":"Not jolly");
        }
	}
}